### PR TITLE
fix(indexer): Add join timeout for indexer parallel consumer

### PR DESCRIPTION
**Pull request by nikhars**
In order to limit the time a consumer can spend during a join, add a parameter to control the join timeout. This is similar to what was done on the snuba consumers in https://github.com/getsentry/snuba/pull/4222

### CODE REVIEW

**Review by volokluev — APPROVED** [review]

**Review by untitaker — DISMISSED** [review]

**Comment by onewland** [comment]
Do we need to be more careful with the join timeout here vs. snuba-generic-metrics? 

This consumer produces messages to the snuba-metrics topic which I assume happens in the join step, and right now there's no logic/support for deduplication. My concern is that on every deploy we could cut off some of the produce step (but not all) and that when we restart we might re-send data to the topic that has already been sent.

File: src/sentry/runner/commands/run.py [hunk 1]
```diff
@@ -599,6 +599,7 @@ def occurrences_ingest_consumer(**options):
 @click.option("--output-block-size", type=int, default=DEFAULT_BLOCK_SIZE)
 @click.option("--ingest-profile", required=True)
 @click.option("--indexer-db", default="postgres")
+@click.option("--join-timeout", type=int, help="Join timeout in seconds.", default=5)
```
Comment (onewland) [inline src/sentry/runner/commands/run.py:602]: if my other comment is right, could we make the default something a little more permissive? something like 60-120s?

**Comment by untitaker** [comment]
it seems to me that we are missing a join()-related bugfix from arroyo that will be shipped here: https://github.com/getsentry/sentry/pull/50316

lets wait and see if that alone improves things

**Comment by nikhars** [comment]
> it seems to me that we are missing a join()-related bugfix from arroyo that will be shipped here: #50316
> 
> lets wait and see if that alone improves things

Ok. Going to wait for that PR to merge.

**Review by lynnagara — APPROVED** [review]

**Comment by lynnagara** [comment]
In relation to @onewland comment, it looks like this was already applying the 5 seconds here before this change https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_metrics/consumers/indexer/multiprocess.py#L95-L96

**Review by lynnagara — APPROVED** [review]

**Pull request merged**